docs/getting-started/part-3/custom-webservice.md

```diff
@@ -0,0 +1,170 @@
+```
+ext-departures {
+  type = CustomWebserviceDataObject
+  baseUrl = "https://opensky-network.org/api/flights/departure"
+  nRetry = 5
+  queryParameters = [{
+    airport="LSZB"
+    begin=1639609200
+    end=1639868400
+  },{
+    airport="EDDF"
+    begin=1639609200
+    end=1639868400
+  }]
+  timeouts {
+    connectionTimeoutMs=200000
+    readTimeoutMs=200000
+  }
+}
+```
+
+  
+The Configuration for this new `ext-departures` includes the type of the Data Object, the base url, where we can fetch the departures, the number of retries, a list of query parameters and timeout options. To have more flexibility we can configure the query parameters now as options instead defining them in the query string. The connection timeout corresponds to the time we wait until the connection is established and the read timeout equals the time we wait until the webservice responds after the request has been submitted. If for example the request cannot be answered in the time configured, we try to automatically resend the request. How many times a failed request will be resend, is controlled by the nRetry parameter.
+
+:::warning
+  For the *begin* and *end* you should **not** choose an interval that is larger than a week. Otherwise the webservice will not respond. As the configuration must be as a unix timestamp, have a look at this [website](https://www.unixtimestamp.com/).
+:::
+
+To make the Data Object of type `CustomWebserviceDataObject` available the following three files had to be included in our project:  
+  - ./src/scala/io/smartdatalake/workflow/dataobject/CustomWebserviceDataObject.scala
+  - ./src/scala/io/smartdatalake/util/webservice/ScalaCustomWebserviceClient.scala
+  - ./src/scala/io/smartdatalake/util/webservice/WebserviceMethod.scala
+
+In this part we will work exclusively with the `CustomWebserviceDataObject.scala` file.
+
+## Define Action
+In the config we change again only one action, namely:
+```
+download-departures {
+  type = CopyAction
+  inputId = ext-departures
+  outputId = stg-departures
+  metadata {
+    feed = download-departures
+  }
+}
+```
+The type is no longer `FileTransferAction`, but instead `CopyAction`. We also changed the feed name. For this part we are mainly interested exectuting this action.
+
+## Try it out
+Re-Build the docker image to update the Scala files and then execute this specific feed using the command below
+```
+  docker build -t smart-data-lake/gs1 .
+  docker run --rm -v ${PWD}/data:/mnt/data -v ${PWD}/config:/mnt/config smart-data-lake/gs1:latest --config /mnt/config --feed-sel download-departures
+```
+Nothing should have changed. You should again receive the data as json files in the corresponding `stg-departures` folder. But except of receiving the departures for only one airport the Data Object returns the departures for all configured airports. In this specific case this would be *LSZB* and *EDDF* with the corresponding time window.
```

Comment:
Its not so smart to write the Spark DataFrame of our CustomWebserviceDataObject again in Json Format, as we loose the schema definition in Json Format. I suggest to remove stg-departure, and write directly into a DeltaLakeTableDataObject in int-layer.

Reply:
@hiltpold : this should also be done, right?

Reply:
@pgruetter: It's implemented. Please check the newest version.